src/main/java/com/redhat/devtools/intellij/commonuitest/fixtures/mainidewindow/toolwindowspane/AbstractToolWinPane.java

```diff
@@ -0,0 +1,156 @@
+        return togglePane(ButtonLabels.MAVEN_STRIPE_BUTTON_LABEL, MavenBuildToolPane.class, true);
+    }
+
+    /**
+     * Close maven build tool pane
+     */
+    public void closeMavenBuildToolPane() {
+        togglePane(ButtonLabels.MAVEN_STRIPE_BUTTON_LABEL, MavenBuildToolPane.class, false);
+    }
+
+    /**
+     * Open gradle build tool pane
+     *
+     * @return the Gradle Build Tool Pane fixture
+     */
+    public GradleBuildToolPane openGradleBuildToolPane() {
+        return togglePane(ButtonLabels.GRADLE_STRIPE_BUTTON_LABEL, GradleBuildToolPane.class, true);
+    }
+
+    /**
+     * Close gradle build tool pane
+     */
+    public void closeGradleBuildToolPane() {
+        togglePane(ButtonLabels.GRADLE_STRIPE_BUTTON_LABEL, GradleBuildToolPane.class, false);
+    }
+
+    /**
+     * Create fixture for the Stripe button
+     *
+     * @param label        label text of the stripe button
+     * @param isPaneOpened true if the pane is already opened
+     * @return fixture for the Stripe button
+     */
+    public JButtonFixture stripeButton(String label, boolean isPaneOpened) {
+        if (isPaneOpened) {
+            if (label.equals(ButtonLabels.MAVEN_STRIPE_BUTTON_LABEL) || label.equals(ButtonLabels.GRADLE_STRIPE_BUTTON_LABEL)) {
+                return button(byXpath("//div[@disabledicon='toolWindow" + label + ".svg']"), Duration.ofSeconds(2));
+            } else if (label.equals(ButtonLabels.PROJECT_STRIPE_BUTTON_LABEL)) {
+                return button(byXpath("//div[@tooltiptext='Project']"), Duration.ofSeconds(2));
+            }
+        }
+        return button(byXpath("//div[@text='" + label + "']"), Duration.ofSeconds(2));
+    }
+
+    private <T extends Fixture> T togglePane(String label, Class<T> fixtureClass, boolean openPane) {
+        if ((!isPaneOpened(fixtureClass) && openPane)) {
+            clickOnStripeButton(label, false);
+            return find(fixtureClass, Duration.ofSeconds(10));
+        } else if (isPaneOpened(fixtureClass) && !openPane) {
+            clickOnStripeButton(label, true);
+        }
+        return null;
+    }
+
+    private boolean isPaneOpened(Class<? extends Fixture> fixtureClass) {
+        AbstractToolWinPane toolWinPane;
+        if (UITestRunner.getIdeaVersionInt() >= 20221) {
+            toolWinPane = remoteRobot.find(ToolWindowPane.class, Duration.ofSeconds(10));
+        } else {
+            toolWinPane = remoteRobot.find(ToolWindowsPane.class, Duration.ofSeconds(10));
+        }
+
+        try {
+            toolWinPane.find(fixtureClass, Duration.ofSeconds(10));
+            return true;
+        } catch (WaitForConditionTimeoutException e) {
+            return false;
+        }
+    }
+
+    private void clickOnStripeButton(String label, boolean isPaneOpened) {
+        waitFor(Duration.ofSeconds(30), Duration.ofSeconds(2), "The '" + label + "' stripe button is not available.", () -> isStripeButtonAvailable(label, isPaneOpened));
```

Comment:
I would to understand this wait a bit more. How many times and for how long it will be executing its condition (isStripeButtonAvailable function)? Given that isStripeButtonAvailable executes find with 10 sec timeout. I am curious what will happen and how long it will take to evaluate this wait if ToolWindow(s)Pane does not appear?

Reply:
This must be an older code I just moved from one class to another. Even I needed to take a look at it a after a short period of time I must say that the 10 second timeout is too much -> decreased to 2 seconds.